pkg/epp/requestcontrol/dag.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package requestcontrol
+
+import (
+	"errors"
+	"slices"
+)
+
+// buildDAG builds a dependency graph among data preparation plugins based on their
+// produced and consumed data keys.
+func buildDAG(plugins []PrepareDataPlugin) map[string][]string {
+	dag := make(map[string][]string)
+	for _, plugin := range plugins {
+		dag[plugin.TypedName().String()] = []string{}
+	}
+	// Create dependency graph as a DAG.
+	for i := range plugins {
+		for j := range plugins {
+			if i == j {
+				continue
+			}
+			// Check whether plugin[i] produces something consumed by plugin[j]. In that case, j depends on i.
+			if plugins[i].Produces() != nil && plugins[j].Consumes() != nil {
+				// For all the keys produced by plugin i, check if plugin j consumes any of them.
+				// If yes, then j depends on i.
+				for producedKey := range plugins[i].Produces() {
+					// If plugin j consumes the produced key, then j depends on i. We can break after the first match.
```

Comment:
@rahulgurnani  What happens if a plugin is dependent on information produced by more than one plugin?